src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java

```diff
@@ -404,13 +415,23 @@ else if (dateOut.isBefore(ad.getAdmDate()) || dateOut.equals(ad.getAdmDate())) {
 									DateTimeFormatter.ISO_LOCAL_DATE.format(invalidEnd))));
 				}
 			}
-
+			List<Disease> diseaseIpdOuts1 = diseaseManager.getDiseaseIpdOut().stream().filter(dis -> dis == admission.getDiseaseOut1()).collect(Collectors.toList());
+			List<Disease> diseaseIpdOuts2 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut2()).collect(Collectors.toList());
+			List<Disease> diseaseIpdOuts3 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut3()).collect(Collectors.toList());
 			if (admission.getDiseaseOut1() == null && admission.getDisDate() != null) {
 				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.pleaseselectatleastfirstdiagnosisout.msg")));
 			} else if (admission.getDiseaseOut1() != null && admission.getDisDate() == null) {
 				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.pleaseinsertadischargedate.msg")));
 			}
-
+			if (admission.getDiseaseOut1() != null && diseaseIpdOuts1.size() > 0) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout1isnotallowed.msg")));
+			}
+			if (admission.getDiseaseOut2() != null && diseaseIpdOuts2.size() > 0) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout2isnotallowed.msg")));
+			}
+			if (admission.getDiseaseOut3() != null && diseaseIpdOuts3.size() > 0) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout3isnotallowed.msg")));
+			}
```

Comment:
With the previous suggestion we have that the filtered `diseaseIpdOutsX` list is empty if the `admission.getDiseaseOutX()` is `null` or not found in the `allowedDiseaseIpdOut` list, so this part changes to:

```suggestion
			if (admission.getDiseaseOut1() != null && diseaseIpdOuts1.isEmpty()) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout1isnotallowed.msg")));
			}
			if (admission.getDiseaseOut2() != null && diseaseIpdOuts2.isEmpty()) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout2isnotallowed.msg")));
			}
			if (admission.getDiseaseOut3() != null && diseaseIpdOuts3.isEmpty()) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout3isnotallowed.msg")));
			}
```

But we can optimize further by checking `admission.getDiseaseOutX() != null` first, and the check the stream object together, for instance:

```
			// get list of allowed disease and single diseases only once
			List<Disease> allowedDiseaseIpdOut = diseaseManager.getDiseaseIpdOut();
			Disease diseaseOut1 = admission.getDiseaseOut1();
			Disease diseaseOut2 = admission.getDiseaseOut2();
			Disease diseaseOut3 = admission.getDiseaseOut3();
			
			// if diseaseOut1  is not null then we check if is within the allowed ones, otherwise error
			if (diseaseOut1 != null && allowedDiseaseIpdOut.stream().filter(dis -> dis.equals(diseaseOut1)).collect(Collectors.toList()).isEmpty()) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout1isnotallowed.msg")));
			}

			// repeat for 2
			if (diseaseOut2 != null && allowedDiseaseIpdOut.stream().filter(dis -> dis.equals(diseaseOut2)).collect(Collectors.toList()).isEmpty()) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout2isnotallowed.msg")));
			}

			// repeat for 3
			if (diseaseOut3 != null && allowedDiseaseIpdOut.stream().filter(dis -> dis.equals(diseaseOut3)).collect(Collectors.toList()).isEmpty()) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout3isnotallowed.msg")));
			}
```